cpp/src/hash/concurrent_unordered_map.cuh

```diff
@@ -93,14 +93,19 @@ union pair_packer<pair_type, std::enable_if_t<is_packable<pair_type>()>> {
  * TODO:
  *  - add constructor that takes pointer to hash_table to avoid allocations
  */
-template <typename Key, typename Element, typename Hasher = default_hash<Key>,
+template <typename Key, typename Element,
+          typename Hasher = default_hash<Key>,
           typename Equality = equal_to<Key>,
-          typename Allocator = default_allocator<thrust::pair<Key, Element>>>
+          typename Allocator = default_allocator<thrust::pair<Key, Element>>,
+          typename FindHasher = Hasher,
```

Comment:
Okay, don't hate me, but I'm going to reverse what I said before. I think it makes more sense to allow passing the key equality operator to insert/find separately.

Reply:
Reverted to the other approach.